lib/msal-common/src/telemetry/performance/PerformanceClient.ts

```diff
@@ -104,6 +122,102 @@ export abstract class PerformanceClient implements IPerformanceClient {
         return {} as IPerformanceMeasurement;
     }
 
+    /**
+     * Sets pre-queue time by correlation Id
+     * 
+     * @abstract
+     * @param {PerformanceEvents} eventName 
+     * @param {string} correlationId 
+     * @returns
+     */
+    abstract setPreQueueTime(eventName: PerformanceEvents, correlationId?: string): void;
+
+    /**
+     * Gets map of pre-queue times by correlation Id
+     * 
+     * @param {PerformanceEvents} eventName 
+     * @param {string} correlationId 
+     * @returns {number}
+     */
+    getPreQueueTime(eventName: PerformanceEvents, correlationId: string): number | void {
+        const preQueueTimesByEvents = this.preQueueTimeByCorrelationId.get(correlationId);
+
+        if (!preQueueTimesByEvents) {
+            this.logger.trace(`PerformanceClient.getPreQueueTime: no pre-queue times found for correlationId: ${correlationId}, unable to add queue measurement`);
+            return;
+        } else if (!preQueueTimesByEvents.get(eventName)) {
+            this.logger.trace(`PerformanceClient.getPreQueueTime: no pre-queue time found for ${eventName}, unable to add queue measurement`);
+            return;
+        }
+
+        return preQueueTimesByEvents.get(eventName);
+    }
+
+    /**
+     * Calculates the difference between current time and time when function was queued.
+     * Note: It is possible to have 0 as the queue time if the current time and the queued time was the same.
+     * 
+     * @param {number} preQueueTime 
+     * @param {number} currentTime 
+     * @returns {number}
+     */
+    calculateQueuedTime(preQueueTime: number, currentTime: number): number {        
+        if (preQueueTime < 1) {
+            this.logger.trace(`PerformanceClient: preQueueTime should be a positive integer and not ${preQueueTime}`);
+            return 0;
+        }
+
+        if (currentTime < 1) {
+            this.logger.trace(`PerformanceClient: currentTime should be a positive integer and not ${currentTime}`);
+            return 0;
+        }
+
+        if (currentTime < preQueueTime) {
+            this.logger.trace("PerformanceClient: currentTime is less than preQueueTime, check how time is being retrieved");
+            return 0;
+        }
+
+        return currentTime-preQueueTime;
```

Comment:
Please ensure that this is not negative.  Yes, that can happen.  Might be worth tracing and setting this to 0 in that case.

Reply:
I think I addressed this in lines 175-178 above, but let me know if you have a better way of handling it.